### PR TITLE
ART-10508: add Jira and PR creation logic with submitRequest button

### PR DESCRIPTION
**Pull request by DennisPeriquet**
This calls the new /git_jira_api endpoint introduced in [this PR](https://github.com/openshift-eng/art-dashboard-server/pull/150) in the art-ui-server backend to create the PR and Jira for image add requests.

That PR must merge first.

### CODE REVIEW

**Comment by DennisPeriquet** [comment]
/hold

This cannot merge until the `/git_jira_api` is merged, up and functional.

File: components/self-service/new-content-form.tsx [hunk 1]
```diff
@@ -116,7 +116,12 @@ export default function NewContentForm({ onSubmit, defaultValues }: { onSubmit?:
       setValue("deliveryRepoApplicationCategories", new Set(newCategories));
       setSelectedAppCategories(newCategories);
     }
-};
+  };
+
+  const [userImageReleaseVersion, setUserImageReleaseVersion] = useState('openshift-4.17')
```
Comment (ashwindasr) [inline components/self-service/new-content-form.tsx:121]: Can we dynamically get the highest branch on `ocp-build-data`, like `openshift-4.19` at this point of time?

File: components/self-service/new-content-done.tsx [hunk 2]
```diff
@@ -174,19 +224,20 @@ export default function NewContentDone() {
...
-      >
-        Create Jira
-      </Button>
+      {!isSubmitted && (
+        <Button
+          variant="contained"
+          onClick={handleSubmitRequest}
+          sx={{ mt: 1, mr: 1 }}
+        >
+          SubmitRequest
```
Comment (ashwindasr) [inline components/self-service/new-content-done.tsx:238]:
```suggestion
          Submit Request
```
nit: add a space between the words

File: components/self-service/new-content-done.tsx [hunk 3]
```diff
@@ -174,19 +224,20 @@ export default function NewContentDone() {
         Follow these steps:
       </Typography>
       <ul>
-        <li>If the above looks good, login to Jira <a href="https://issues.redhat.com/login.jsp?os_destination=%2Fdefault.jsp" target="_blank" rel="noopener noreferrer">here</a> (a separate tab will open) then click the "Create Jira" button below</li>
-        <li>In the "Create Issue" page, set the "Reporter" field as your UserId and click the "Create" button at the bottom</li>
-        <li>Share the ticket to <strong>@release-artists</strong> on <strong>#forum-ocp-art</strong> on Slack</li>
+        <li>If the above looks good, click SubmitRequest and a Jira and PR will be created</li>
```
Comment (ashwindasr) [inline components/self-service/new-content-done.tsx:227]:
```suggestion
        <li>If the above looks good, click Submit Request and a Jira and PR will be created</li>
```
same nit

**Review by ashwindasr — CHANGES_REQUESTED** [review]
Amazing work @DennisPeriquet! Thanks for working on this!

Minor nits in this PR, but the major issue is that we should restrict the number of times the JIRA and PR creation requests are sent it (i.e. to only once, as mentioned in [Slack](https://redhat-internal.slack.com/archives/C07T08DJWRE/p1731077012782389))

File: components/self-service/new-content-form.tsx (hunk 1 above)
Comment (DennisPeriquet) [reply components/self-service/new-content-form.tsx]: this is now implemented

File: components/self-service/new-content-done.tsx (hunk 2 above)
Comment (DennisPeriquet) [reply components/self-service/new-content-done.tsx]: fixed

**Comment by DennisPeriquet** [comment]
re: the case where the api call to create the PR and Jira fails, ... if this happens, we now keep the "Submit Request" button up so that user can re-click the button and retry without having to re-enter all of the values they set earlier.

**Review by ashwindasr — APPROVED** [review]
Server [PR](https://github.com/openshift-eng/art-dashboard-server/pull/150) has merged

**Pull request merged**